### PR TITLE
PR: Ignore warning generated by ipykernel.datapub

### LINKED ISSUE #5591: Variable explorer broken because of DeprecationWarning in ipykernel
## Description of your problem
When I try to access anything in the variable explorer I get the message:

DeprecationWarning: ipykernel.datapub is deprecated. It has moved to ipyparallel.datapub

**What steps will reproduce the problem?**

1. open spyder
2. run a script 
3. click on anything in the variable explorer

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

## Versions and main components

* Spyder Version: 3.2.4
* Python Version: 3.5
* Qt Version: 5.6.2
* PyQt Version: 5.6.0
* Operating system: Mac

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:
IPython >=4.0    :  5.3.0 (OK)
cython >=0.21    :  0.25.2 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
numpy >=1.7      :  1.13.1 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
psutil >=0.3     :  5.2.2 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.5.6 (OK)
sympy >=0.7.3    :  1.0 (OK)

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #5591

### CODE REVIEW

File: spyder/utils/site/sitecustomize.py [hunk 1]
```diff
@@ -280,6 +280,14 @@ def __init__(self, *args, **kwargs):
         TestProgram.__init__(self, *args, **kwargs)
 unittest.main = IPyTesProgram
 
+# Filter warnings that appear for ipykernel wheninteracting with
```
Comment (ccordoba12) [inline spyder/utils/site/sitecustomize.py]: wheninteracting -> when interacting

File: spyder/utils/site/sitecustomize.py [hunk 2]
```diff
@@ -280,6 +280,14 @@ def __init__(self, *args, **kwargs):
         TestProgram.__init__(self, *args, **kwargs)
 unittest.main = IPyTesProgram
 
+# Filter warnings that appear for ipykernel wheninteracting with
+# the Variable explorer (i.e trying to see a variable)
+# Fixes Issue 5591
+warnings.filterwarnings(action='ignore', category=DeprecationWarning,
+                        module='ipykernel.datapub',
+                        message=".*ipykernel.datapub is deprecated. "
```
Comment (ccordoba12) [inline spyder/utils/site/sitecustomize.py]: I think a single line is enough here. So it should be

    message=".*ipykernel.datapub is deprecated.*"

**Comment by ccordoba12** [comment]
@JoshKaron, could you test that this PR fixes your problem? You can copy the lines added by @dalthviz to your `sitecustomize` file (present in your Spyder installed package) and try again.

**Comment by arunel** [comment]
@ccordoba12, I also had this issue. I applied the fix added by @dalthviz and it worked, no more warnings! Thanks.

**Comment by ccordoba12** [comment]
Fantastic! Thanks a lot for the confirmation!

Since we're not able to reproduce this problem in our side, your help is really valuable.

**Pull request merged**